src/Graphing.hs

```diff
@@ -61,6 +62,14 @@ filter f gr = gr { graphingDirect = direct', graphingAdjacent = adjacent' }
 empty :: Graphing ty
 empty = Graphing S.empty AM.empty
 
+-- | Strip a given item from the direct set, promote its immediate children to direct items
+stripRoot :: Ord ty => Graphing ty -> Graphing ty
+stripRoot gr = gr { graphingDirect = direct' }
+  where
+  topLayer = graphingDirect gr
+  newDirects = S.unions $ map (flip AM.postSet $ graphingAdjacent gr) $ S.toList topLayer
```

Comment:
[style] another way to write `(flip AM.postSet $ graphingAdjacent gr)` is to use postSet infix:
```
(`AM.postSet` graphingAdjacent gr)
```

Reply:
That hurts my brain.  Infix makes sense to me, but infix as a pipeline inside of parens is really hard for me to put together mentally without more experience seeing it.

I'm going to leave this one as-is.